gcloud-java-logging/src/main/java/com/google/cloud/logging/AsyncLoggingHandler.java

```diff
@@ -0,0 +1,97 @@
+
+import java.util.List;
+import java.util.logging.Filter;
+import java.util.logging.Formatter;
+import java.util.logging.Logger;
+import java.util.logging.SimpleFormatter;
+
+/**
+ * A logging handler that asynchronously outputs logs generated with
+ * {@link java.util.logging.Logger} to Stackdriver Logging.
+ *
+ * <p>Java logging levels (see {@link java.util.logging.Level}) are mapped to the following Google
+ * Stackdriver Logging severities:
+ *
+ * <table summary="Mapping of Java logging level to Stackdriver Logging severities">
+ * <tr><th width="50%">Java Level</th><th>Stackdriver Logging Severity</th></tr>
+ * <tr><td>SEVERE</td><td>ERROR</td></tr>
+ * <tr><td>WARNING</td><td>WARNING</td></tr>
+ * <tr><td>INFO</td><td>INFO</td></tr>
+ * <tr><td>CONFIG</td><td>INFO</td></tr>
+ * <tr><td>FINE</td><td>DEBUG</td></tr>
+ * <tr><td>FINER</td><td>DEBUG</td></tr>
+ * <tr><td>FINEST</td><td>DEBUG</td></tr>
+ * </table>
+ *
+ * <p>Original Java logging levels are added as labels (with {@code levelName} and
+ * {@code levelValue} keys, respectively) to the corresponding Stackdriver Logging {@link LogEntry}.
+ * You can read entry labels using {@link LogEntry#labels()}. To use logging levels that correspond
+ * to Stackdriver Logging severities you can use {@link LoggingLevel}.
+ *
+ * <p><b>Configuration</b>: By default each {@code AsyncLoggingHandler} is initialized using the
+ * following {@code LogManager} configuration properties (that you can set in the
+ * {@code logging.properties} file. If properties are not defined (or have invalid values) then the
+ * specified default values are used.
+ * <ul>
+ * <li>{@code com.google.cloud.logging.AsyncLoggingHandler.log} the log name (defaults to
+ *     {@code java.log}).
+ * <li>{@code com.google.cloud.logging.AsyncLoggingHandler.level} specifies the default level for
+ *     the handler (defaults to {@code Level.INFO}).
+ * <li>{@code com.google.cloud.logging.AsyncLoggingHandler.filter} specifies the name of a
+ *     {@link Filter} class to use (defaults to no filter).
+ * <li>{@code com.google.cloud.logging.AsyncLoggingHandler.formatter} specifies the name of a
+ *     {@link Formatter} class to use (defaults to {@link SimpleFormatter}).
+ * <li>{@code com.google.cloud.logging.AsyncLoggingHandler.flushSize} specifies the maximum size of
+ *     the log buffer. Once reached, logs are transmitted to the Stackdriver Logging service
+ *     (defaults to 1).
+ * <li>{@code com.google.cloud.logging.AsyncLoggingHandler.flushLevel} specifies the flush log
+ *     level. When a log with this level is published, logs are transmitted to the Stackdriver
+ *     Logging service (defaults to {@link LoggingLevel#ERROR}).
+ * </ul>
+ *
+ * <p>To add a {@code LoggingHandler} to an existing {@link Logger} and be sure to avoid infinite
+ * recursion when logging, use the {@link #addHandler(Logger, LoggingHandler)} method.
+ */
+public class AsyncLoggingHandler extends LoggingHandler {
```

Comment:
You mean mentioning in [this paragraph](https://github.com/GoogleCloudPlatform/gcloud-java/pull/1117/files/99a6ae1354a36eb83f13e5802b3fcb73df7a08d0#diff-f653e035fa60c430e457a3ce1150169dR72) that the handler can also be registered via {@logging.properties}? Yes I agree we should mention it, I'll fix this.